solc/CommandLineInterface.cpp

```diff
@@ -692,7 +694,12 @@ void CommandLineInterface::compile()
 {
 	solAssert(CompilerInputModes.count(m_options.input.mode) == 1);
 
-	m_compiler = make_unique<CompilerStack>(m_fileReader.reader());
+	auto modelChecker = m_options.modelChecker.settings;
+	auto universalCallback{UniversalCallback(m_fileReader, SMTSolverCommand("eld"))};
+	if (modelChecker.solvers.eld)
+		m_compiler = make_unique<CompilerStack>(universalCallback.callback());
```

Comment:
You can put the callback inside the `if` branch and avoid making a copy of the reader when eld is not used.

Reply:
Somehow we can't. That's what we had before and what causes crashes

Reply:
@cameel if we put the callback inside the if, `universalCallback` goes out of scope right after.